agent/structs/structs.go

```diff
@@ -70,6 +70,42 @@ const (
 	FederationStateRequestType                  = 30
 )
 
+// requestTypeStrings is used for snapshot enhance
+// any new request types added must be placed here
+var requestTypeStrings = map[MessageType]string{
+	0:  "Register",
+	1:  "Deregister",
+	2:  "KVS",
+	3:  "Session",
+	4:  "ACL", // DEPRECATED (ACL-Legacy-Compat)
+	5:  "Tombstone",
+	6:  "CoordinateBatchUpdate",
+	7:  "PreparedQueryRequest",
+	8:  "TxnRequest",
+	9:  "AutopilotRequest",
+	10: "Area",
+	11: "ACLBootstrap",
+	12: "Intention",
+	13: "ConnectCA",
+	14: "ConnectCAProviderState",
+	15: "ConnectCAConfig", // FSM snapshots only.
+	16: "Index",           // FSM snapshots only.
+	17: "ACLTokenSet",
+	18: "ACLTokenDelete",
+	19: "ACLPolicySet",
+	20: "ACLPolicyDelete",
+	21: "ConnectCALeaf",
+	22: "ConfigEntry",
+	23: "ACLRoleSet",
+	24: "ACLRoleDelete",
+	25: "ACLBindingRuleSet",
+	26: "ACLBindingRuleDelete",
+	27: "ACLAuthMethodSet",
+	28: "ACLAuthMethodDelete",
+	29: "ChunkingState",
+	30: "FederationState",
```

Comment:
Sorry, but I just added `31` today: https://github.com/hashicorp/consul/blob/master/agent/structs/structs.go#L71

Also remember you'll have to do some trickery to inject a few more items into this map from ent code in an init() block or something since there are a few ent-only raft operations.